apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -1077,6 +1078,12 @@ where
                 edges.push(edge);
                 edge_triggers.push(Arc::new(trigger));
                 edge_conditions.push(condition_path_tree);
+                if self.graph.is_cross_subgraph_edge(new_edge)? {
+                    last_subgraph_entering_edge_info = Some(SubgraphEnteringEdgeInfo {
+                        index: self.edges.len() - 1,
+                        conditions_cost: condition_cost,
+                    });
+                }
```

Comment:
This logic (and the similar statement below) more accurately reflects the JS code. The old logic would have `last_subgraph_entering_edge_info` be `None` if `is_cross_subgraph_edge` returned `false`, but the JS code would not.

Reply:
just wondering should we also set it to `None` otherwise?

Reply:
No, it should be `Some`. We are checking to see if the new edge crosses subgraphs. If so, the index needs to be updated. If the new edge doesn't cross a subgraph, we still need to track the info of the last entering edge.

Reply:
@TylerBloom
Good catch! We're indeed setting `last_subgraph_entering_edge_info` to `None` when we shouldn't be.